service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -124,25 +129,65 @@ public PolicyEvaluationResult evalPolicy(final String uri, final String subjectI
 
         if (allPolicySets.isEmpty()) {
             return new PolicyEvaluationResult(Effect.NOT_APPLICABLE);
-        } else if (allPolicySets.size() > 1) {
-            LOGGER.error("Found more than one policy set during policy evaluation. Subject: " + subjectIdentifier
-                    + ", Resource: " + uri);
-            throw new IllegalArgumentException("More than one policy set exists for this zone. "
-                    + "Remove unnecessary policy sets using DELETE /policy-set/{id} and resubmit request.");
-        } else {
-            // NOTE: When multiple policy sets are supported, this code needs to
-            // delegate to a "InterPolicySetDecisionAggregator"
-            PolicyEvaluationResult result = evalPolicySet(allPolicySets.get(0), subjectIdentifier, uri, action,
+        }
+
+        List<PolicySet> filteredPolicySets = filterPolicySetsByPriority(subjectIdentifier, uri, allPolicySets,
+                policySetsPriority);
+        PolicyEvaluationResult result = new PolicyEvaluationResult(Effect.NOT_APPLICABLE);
+
+        for (PolicySet policySet : filteredPolicySets) {
+            result = evalPolicySet(policySet, subjectIdentifier, uri, action,
                     supplementalResourceAttributes, supplementalSubjectAttributes);
+            if (result.getEffect() != Effect.NOT_APPLICABLE) {
```

Comment:
If a request specifies multiple policy sets that need to be evaluated in a specific order, ACS should not give up after the result of evaluating the first policy set is indeterminate. Instead, it should move on to the next one and then the next one until there is a clear evaluation result or until the specified policy sets are exhausted.